### PR TITLE
[4.0] Rules field prepared statements

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds prepared statements to `Joomla\CMS\Form\Field\RulesField`.

### Testing Instructions

Go to Global Configuration. Switch to permissions tab. Inspect permission field.
Go to some component's configuration. Switch to permissions tab. Inspect the field.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: libraries/src/Form/Field/RulesField.php [hunk 1]
```diff
@@ -179,23 +180,24 @@ protected function getInput()
 
 		// Get the asset id.
 		// Note that for global configuration, com_config injects asset_id = 1 into the form.
-		$this->assetId       = $this->form->getValue($assetField);
-		$this->newItem       = empty($assetId) && $this->isGlobalConfig === false && $section !== 'component';
+		$this->assetId       = (int) $this->form->getValue($assetField);
```
Comment (Quy) [inline libraries/src/Form/Field/RulesField.php]:
```suggestion
		$this->assetId = (int) $this->form->getValue($assetField);
```

File: libraries/src/Form/Field/RulesField.php [hunk 2]
```diff
@@ -179,23 +180,24 @@ protected function getInput()
...
+		$this->newItem       = empty($this->assetId) && $this->isGlobalConfig === false && $section !== 'component';
```
Comment (Quy) [inline libraries/src/Form/Field/RulesField.php]:
```suggestion
		$this->newItem = empty($this->assetId) && $this->isGlobalConfig === false && $section !== 'component';
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b274338a2dd4933c5f98424e874e59cd669ff27e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26082">issues.joomla.org/tracker/joomla-cms/26082</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on b274338a2dd4933c5f98424e874e59cd669ff27e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26082">issues.joomla.org/tracker/joomla-cms/26082</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26082">issues.joomla.org/tracker/joomla-cms/26082</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**